doc/source/examining/Loading_Generic_Array_Data.ipynb

```diff
@@ -688,6 +686,47 @@
     "slc.show()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "## Species fields"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "One can also supply species fields to a stream dataset, in the form of mass fractions. These will then be used to generate derived fields for mass, number, and nuclei densities of the separate species. The naming conventions for the mass fractions should correspond to the format specified in [the yt documentation for species fields](https://yt-project.org/doc/analyzing/fields.html#species-fields)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "data = {\n",
+    "    \"density\": (arr, \"g/cm**3\"),\n",
```

Comment:
there's a `NameError` here (`arr` is undefined)

Reply:
that's weird because it's using the `arr` created at the top of the notebook, but I guess that it's better to have it self-contained

Reply:
Fixed in notebook